sequencer.c

```diff
@@ -4635,9 +4635,13 @@ static int make_script_with_merges(struct pretty_print_context *pp,
 		else
 			strbuf_addbuf(&label, &oneline);
 
+		/* sanitize label */
 		for (p1 = label.buf; *p1; p1++)
 			if (isspace(*p1))
 				*(char *)p1 = '-';
+			/* check if non alphanumeric ascii character */
```

Comment:
I think we should also explain the `*p1 & 0x80` part, i.e. something like this instead:

```c
/*
 * Replace non-alphanumeric characters by dashes, as they might be illegal in
 * file names (and hence in ref names).
 *
 * Note that we retain non-ASCII UTF-8 characters (identified via the most
 * significant bit). They should be all acceptable in file names.
 */
```

Reply:
I added in that comment as you're probably right about that

Reply:
I think that "check if non alphanumeric ascii character" is a bit misleading: we want to retain _alpha-numeric ASCII* characters, but that comment tells me that we want to retain *non-*alpha-numeric ones.

Reply:
Actually, we want to retain non-ASCII UTF-8 characters, so what I said does not make too much sense. But that one-liner comment is still incomplete... I fixed it by using the verbose comment (after editing it slightly).